src/components/payroll/PayrollPaymentFilesSearcher.js

```diff
@@ -0,0 +1,111 @@
+  Searcher,
+  useModulesManager,
+  useTranslations,
+} from '@openimis/fe-core';
+import {
+  DEFAULT_PAGE_SIZE, MODULE_NAME,
+  ROWS_PER_PAGE_OPTIONS, PAYROLL_PAYMENT_FILE_STATUS,
+} from '../../constants';
+import { fetchPayrollPaymentFiles } from '../../actions';
+import downloadPayroll from '../../utils/export';
+
+function PayrollPaymentFilesSearcher({
+  fetchingPayrollFiles,
+  fetchedPayrollFiles,
+  errorPayrollFiles,
+  files,
+  pageInfo,
+  totalCount,
+  fetchPayrollPaymentFiles,
+  payrollUuid,
+}) {
+  const modulesManager = useModulesManager();
+  const { formatMessage, formatMessageWithValues } = useTranslations(MODULE_NAME, modulesManager);
+
+  const headers = () => [
+    'payrollPaymentFile.fileName',
+    'payrollPaymentFile.status',
+    'payrollPaymentFile.download',
+  ];
+
+  const defaultFilters = () => {
+    const filters = {
+      isDeleted: {
+        value: false,
+        filter: 'isDeleted: false',
+      },
+    };
+    if (payrollUuid) {
+      filters.payrollId = {
+        value: payrollUuid,
+        filter: `payroll_Id: "${payrollUuid}"`,
+      };
+    }
+    return filters;
+  };
+
+  const download = (payrollId, fileName) => {
+    downloadPayroll(payrollId, fileName, false);
+  };
+
+  const fetchFiles = (params) => fetchPayrollPaymentFiles(modulesManager, params);
+
+  const rowIdentifier = (file) => file.fileName;
+
+  const itemFormatters = () => [
+    (file) => file.fileName,
+    (file) => file.status,
+    (file) => (
+      <Tooltip title={formatMessage('tooltip.delete')}>
+        <IconButton
+          onClick={() => download(payrollUuid, file.fileName)}
+          disabled={file.status !== PAYROLL_PAYMENT_FILE_STATUS.SUCCESS}
+        >
+          <DownloadIcon />
+        </IconButton>
+      </Tooltip>
+    ),
+  ];
+
+  return (
+    <Searcher
+      module="payroll"
+      FilterPane={null}
+      fetch={fetchFiles}
+      items={files}
+      itemsPageInfo={pageInfo}
+      fetchedItems={fetchedPayrollFiles}
+      fetchingItems={fetchingPayrollFiles}
+      errorItems={errorPayrollFiles}
+      tableTitle={formatMessageWithValues('payrollPaymentFilesSearcher.results', { totalCount })}
+      headers={headers}
+      itemFormatters={itemFormatters}
+      rowsPerPageOptions={ROWS_PER_PAGE_OPTIONS}
+      defaultPageSize={DEFAULT_PAGE_SIZE}
+      rowIdentifier={rowIdentifier}
+      defaultFilters={defaultFilters()}
+    />
+  );
+}
+const mapStateToProps = (state) => ({
+  fetchingPayrollFiles: state.payroll.fetchingPayrollFiles,
+  fetchedPayrollFiles: state.payroll.fetchedPayrollFiles,
+  errorPayrollFiles: state.payroll.errorPayrollFiles,
+  files: state.payroll.payrollFiles,
+  pageInfo: state.payroll.payrollFilesPageInfo,
+  totalCount: state.payroll.payrollFilesTotalCount,
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  fetchPayrollPaymentFiles,
+}, dispatch);
+
+export default connect(mapStateToProps, mapDispatchToProps)(PayrollPaymentFilesSearcher);
```

Comment:
Is there any reason why you're using mapStateToProps/mapDispatchToProps? Shouldn't we use useSelector/useDispatch instead?

Reply:
just the matter of copy-paste mechanism. Due to limited time let's keep as it is. @olewandowski1

Reply:
ok then i am keeping it as it is